hs_core/page_processors.py

```diff
@@ -278,6 +288,21 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     maps_key = settings.MAPS_KEY if hasattr(settings, 'MAPS_KEY') else ''
 
+    def group_to_community(grp, communities):
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    grps_member_of = []
+    groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
+    # for each group set group dynamic attributes
+    for g in groups:
+        g.is_user_member = user in g.gaccess.members
+        if g.is_user_member:
+            grps_member_of.append(g)
+
+    comms_member_of = [group_to_community(g, Community.objects.all()) for g in grps_member_of]
```

Comment:
This query should be done in the db and shouldn't be done for all resources and users in all instances

Reply:
This was stop-gap while Alva was on another project and I was mocking something for the customer to demo functionality. Addressed in 1525edfbd